### PR TITLE
kyndryl mainframe connector

### PR DESCRIPTION
**Pull request by RTkyndryl**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,707 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Kyndryl mainframe connector",
+    "version": "1.0.0",
+    "title": "kyndryl"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl I would highly recommend starting with a capital letter "Kyndryl"

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,707 @@
+{
+  "swagger": "2.0",
+  "info": {
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:3]: @RTkyndryl Please make sure to add the contact section inside the info section. 
```
    "contact": {
      "name": "Full name",
      "url": "Contact URL",
      "email": "Contact email address"
    }
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/cb12-policy/getpolicy": {
+      "get": {
+        "tags": [
+          "cb12-policy"
+        ],
+        "operationId": "getPolicy",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl all operationIds should be in Camel Case. please change to "GetPolicy". Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations). 
Please fix all operationIds in the swagger file.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,707 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Kyndryl mainframe connector",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' value must be between '30' and '500' characters.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,707 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/getPolicy_response_200"
+            }
+          }
+        },
+        "summary": "Get policy"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' property is required. please add description under the summary. 
```
"summary": "Get policy",
"description": "Your description"
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,707 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/postCustomerdetailsupd_response_200"
+            }
+          }
+        },
+        "summary": "Post customer details"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' property is required. please add description under the summary. 
```
"summary": "Post customer details",
"description": "Your description"
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,707 @@
...
+                    "CA_PHONE_HOME": "string",
+                    "CA_EMAIL_ADDRESS": "string",
+                    "CA_POLICY_DATA": "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "summary": "Put customer details"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' property is required. please add description under the summary. 
```
"summary": "Put customer details",
"description": "Your description"
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,707 @@
...
+                      "description": "CA_CUSTOMER_NUM"
+                    }
+                  },
+                  "description": "LGCMAREA"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get customer details"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' property is required. please add description under the summary. 
```
"summary": "Get customer details",
"description": "Your description"
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,707 @@
...
+  "paths": {
+    "/cb12-policy/getpolicy": {
+      "get": {
+        "tags": [
+          "cb12-policy"
+        ],
+        "operationId": "getPolicy",
+        "parameters": [
+          {
+            "name": "CUSTOMERNUMBER",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:33]: @RTkyndryl for all parameters The 'description' property is required and the 'x-ms-summary' property is required.
Please add both for each parameter in the swagger file. 
```
"name": "CUSTOMERNUMBER",
"description": "Your desctiption",
"x-ms-summary": "Your summary"
```

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,707 @@
...
+    "/customernumber/Custdetailadd": {
+      "post": {
+        "tags": [
+          "customernumber"
+        ],
+        "operationId": "postCustomerdetailsupd",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "postCustomerdetailsupd_request",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:62]: @RTkyndryl for all parameters The 'x-ms-summary' property is required.
Please add it for each parameter in the swagger file.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,707 @@
...
+    },
+    "/customernumber/Custdetailupd/{num}/{firstname}": {
+      "put": {
+        "tags": [
+          "customernumber"
+        ],
+        "operationId": "putCustomerdetail",
+        "parameters": [
+          {
+            "name": "num",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:91]: @RTkyndryl for all parameters The 'description' property is required and the 'x-ms-summary' property is required.
Please add both for each parameter in the swagger file.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,707 @@
...
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "firstname",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:99]: @RTkyndryl for all parameters The 'description' property is required and the 'x-ms-summary' property is required.
Please add both for each parameter in the swagger file.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,707 @@
...
+    },
+    "/customernumber/custnum/{num}": {
+      "get": {
+        "tags": [
+          "customernumber"
+        ],
+        "operationId": "getCustomerdetail",
+        "parameters": [
+          {
+            "name": "num",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json:160]: @RTkyndryl for all parameters The 'description' property is required and the 'x-ms-summary' property is required.
Please add both for each parameter in the swagger file.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,707 @@
...
+      "in": "header",
+      "name": "Ocp-Apim-Subscription-Key"
+    }
+  },
+  "security": [
+    {
+      "API Key": []
+    }
+  ],
+  "tags": []
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'x-ms-connector-metadata' property is required.
```
 "x-ms-connector-metadata": [
    {
      "propertyName": "Website",
      "propertyValue": ""
    },
    {
      "propertyName": "Privacy policy",
      "propertyValue": ""
    },
    {
      "propertyName": "Categories",
      "propertyValue": ""
    }
  ]
```

File: certified-connectors/KyndrylMainframe/apiProperties.json [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+          "tooltip": "Provide your API Key",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiProperties.json]: @RTkyndryl this is Microsoft standard color and unfortunately it is not allowed to use it. May I ask you please to change it. please follow our requirements [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector).

File: certified-connectors/KyndrylMainframe/readme.md [hunk 16]
```diff
@@ -0,0 +1,35 @@
+# Kyndryl Mainframe Connector
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/readme.md:1]: @RTkyndryl please follow our template for the readme file [here ](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md)and for raw code view [here](https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/certified-connectors/readme.md).

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by RTkyndryl** [comment]
@microsoft-github-policy-service agree company="Kyndryl"

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 17]
```diff
@@ -0,0 +1,740 @@
...
+        "tags": [
+          "customernumber"
+        ],
+        "operationId": "PutCustomerDetail",
+        "parameters": [
+          {
+            "name": "num",
+            "in": "path",
+            "required": true,
+            "type": "string"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl Please add a comma "," at the end of the line 94.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 18]
```diff
@@ -0,0 +1,740 @@
...
+            "description": "Customer Number",
+            "x-ms-summary": "Customer Number",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl Please remove the comma "," at the end of this line 96

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 19]
```diff
@@ -0,0 +1,740 @@
...
+          },
+          {
+            "name": "firstname",
+            "in": "path",
+            "required": true,
+            "type": "string"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl Please add a comma "," at the end of the line 102

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 20]
```diff
@@ -0,0 +1,740 @@
...
+            "description": "Customer First Name",
+            "x-ms-summary": "Customer First Name",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl Please remove the comma "," at the end of this line 104

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 21]
```diff
@@ -0,0 +1,740 @@
...
+        ],
+        "operationId": "GetCustomerDetail",
+        "parameters": [
+          {
+            "name": "num",
+            "in": "path",
+            "required": true,
+            "type": "string", 
+            "description": "Customer Number",
+            "x-ms-summary": "Customer Number",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl Please remove the comma "," at the end of this line 165

File: certified-connectors/KyndrylMainframe/apiProperties.json [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#FEFEFE",
+    "capabilities": [],
+    "publisher": "Ryan Treacy"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiProperties.json]: @RTkyndryl please add the "stackOwner" Property. Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiProperties.json). 
```
"stackOwner": "Name of the owner of the service or product."
```

File: certified-connectors/KyndrylMainframe/readme.md [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+Unlock the value of your data by using this connector to enable low-code developers to build power apps that will unleash new data insights and capabilities across your organization. Work with the thousands of mainframe developers at Kyndryl to innovate and provide better end user experiences against your mainframe data whilst securing it through Kyndryl Bridge and Microsoft Power Platform. 
+
+## Publisher: Publisher's Name
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/readme.md]: @RTkyndryl Please replace the "Publisher's Name" with "Kyndryl"

File: certified-connectors/KyndrylMainframe/readme.md [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+### Create Customer
+This action creates a new customer profile. 
+
+### Get Policy
+This action retrieves a policy related to a customer. 
+
+### Update Customer
+This action updates a specific customers details. 
+
+## Obtaining Credentials
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/readme.md:24]: @RTkyndryl is it possible to provide some links to the page for obtaining the credentials?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: certified-connectors/KyndrylMainframe/apiProperties.json [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiProperties.json]: @RTkyndryl Please add a comma "," at the end of line 20

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 26]
```diff
@@ -0,0 +1,740 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "Unlock the value of your data by using this connector to enable low-code developers to build power apps that will unleash new data insights and capabilities across your organization. Work with the thousands of mainframe developers at Kyndryl to innovate and provide better end user experiences against your mainframe data whilst securing it through Kyndryl Bridge and Microsoft Power Platform. ",
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'description' contains at least one of the restricted words: 'power apps, power automate, logic apps'.  Please remove it from the description.

File: certified-connectors/KyndrylMainframe/apiDefinition.swagger.json [hunk 27]
```diff
@@ -0,0 +1,740 @@
...
+  "security": [
+    {
+      "API Key": []
+    }
+  ],
+  "tags": [], 
+"x-ms-connector-metadata": [
+    {
+      "propertyName": "Website",
+      "propertyValue": "www.kyndryl.com"
```
Comment (Amjed-Ayoub) [inline certified-connectors/KyndrylMainframe/apiDefinition.swagger.json]: @RTkyndryl The 'url' property must be a valid absolute URL. Please change it to "https://www.kyndryl.com"

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @RTkyndryl ,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**